File: dfs_bfs_backtracking/cpp/b2023.cpp
#include <iostream>
#include <math.h>
bool is_prime(const int& N) {
	for (int i = 2; i < int(std::sqrt(N)) + 1; ++i) {
		if (N % i == 0) {
			return false;
		}
	}
	return true;
}

void is_special_prime(const int& N, const int& now_N, const int& temp_num) {
	if (now_N == N) {
		std::cout << temp_num << std::endl;
		return;
	}

	for (int i = 1; i < 10; ++i) {
		if (now_N == 0 && i == 1) {
			continue;
		}

		if (is_prime(temp_num * 10 + i)) {
			is_special_prime(N, now_N + 1, temp_num * 10 + i);
		}
	}
	return;
}

int main() {
	int N;
	std::cin >> N;
	is_special_prime(N, 0, 0);
}

File: dfs_bfs_backtracking/python/b2023.py
def is_prime(number):
    for i in range(2, int(number ** (1 / 2)) + 1):
        if not number % i:
            return False
    return True

def is_special_prime(N, digits, temp_num):
    if N == digits:
        print(temp_num)
        return

    for i in range(10):
        if not digits and (i == 0 or i == 1 or i == 4 or i == 6 or i == 8 or i == 9):
            continue
        new_temp_num = temp_num * 10 + i
        if is_prime(new_temp_num):
            is_special_prime(N, digits + 1, new_temp_num)

    return

N = int(input())
is_special_prime(N, 0, 0)